pkg/controller/template/render.go

```diff
@@ -233,7 +249,12 @@ func generateMachineConfigForName(config *RenderConfig, role, name, path string)
 		return nil, fmt.Errorf("error transpiling ct config to Ignition config: %v", err)
 	}
 
-	return MachineConfigFromIgnConfig(role, name, ignCfg), nil
+	mcfg := MachineConfigFromIgnConfig(role, name, ignCfg)
+	if osUpdatesEnabledForRole(config, role) {
+		mcfg.Spec.OSImageURL = config.OSImageURL
```

Comment:
Let's maybe also log something in the `else` branch here so it's clear why it's not picking it up for some pools?

Reply:
See https://github.com/openshift/machine-config-operator/pull/348